pkg/operator/controller/ingress/deployment.go

```diff
@@ -1123,6 +1123,17 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		)
 	}
 
+	// Add router external certificate environment variable.
+	if routeExternalCertificateEnabled {
+		env = append(env,
+			corev1.EnvVar{Name: "ROUTER_EXTERNAL_CERTIFICATE", Value: "true"},
+		)
+	} else {
+		env = append(env,
+			corev1.EnvVar{Name: "ROUTER_EXTERNAL_CERTIFICATE", Value: "false"},
+		)
+	}
```

Comment:
I think it's better to only add the env variable, if routeExternalCertificateEnabled. If feature-gate is not enabled, then do nothing. At the router side we will check `ROUTER_EXTERNAL_CERTIFICATE` exists or not, and take actions accordingly. 

In that case, we can skip the else statement

Reply:
Done